lib/image-stream/utils.js

```diff
@@ -16,47 +16,30 @@
 
 'use strict';
 
-const _ = require('lodash');
 const Bluebird = require('bluebird');
-const fs = Bluebird.promisifyAll(require('fs'));
-const archiveType = require('archive-type');
+const mime = require('mime-types');
 
 /**
  * @summary Get archive mime type
  * @function
  * @public
  *
- * @param {String} file - file path
+ * @param {String} filename - file path
  * @fulfil {String} - mime type
  * @returns {Promise}
  *
  * @example
- * utils.getArchiveMimeType('path/to/raspberrypi.img.gz').then((mimeType) => {
- *   console.log(mimeType);
- * });
+ * const mimeType = utils.getArchiveMimeType('path/to/raspberrypi.img.gz');
  */
-exports.getArchiveMimeType = (file) => {
-
-  // `archive-type` only needs the first 261 bytes
-  // See https://github.com/kevva/archive-type
-  const ARCHIVE_TYPE_IDENTIFICATION_BYTES_LENGTH = 261;
-
-  return Bluebird.using(fs.openAsync(file, 'r').disposer((fileDescriptor) => {
-    return fs.closeAsync(fileDescriptor);
-  }), (fileDescriptor) => {
-    const BUFFER_START = 0;
-    const chunk = new Buffer(ARCHIVE_TYPE_IDENTIFICATION_BYTES_LENGTH);
+exports.getArchiveMimeType = (filename) => {
+  // NOTE: For some awkward reason `mime-db` (used by `mime-types`)
+  // does not contain a file extension for `.gz`, so we need to test for it here
+  // TODO: Remove this once https://github.com/jshttp/mime-db/pull/68 is merged
+  if (/\.gz$/i.test(filename)) {
+    return 'application/gzip';
+  }
 
-    return fs.readAsync(
-      fileDescriptor,
-      chunk,
-      BUFFER_START,
-      ARCHIVE_TYPE_IDENTIFICATION_BYTES_LENGTH,
-      null
-    ).then(() => {
-      return _.get(archiveType(chunk), [ 'mime' ], 'application/octet-stream');
-    });
-  });
+  return mime.lookup(filename) || 'application/octet-stream';
```

Comment:
> Where does it say that? Looking at the code, the highest offset into the buffer I could find is 261.

Yeah, I was wondering that too! Eventually figured out (by looking through git history) that it's due to https://github.com/sindresorhus/file-type/blob/master/index.js#L175

> So if we use file-type before mime-types, then it will incorrectly detect a DMG as something else.

Not if we filter out anything not in the `[ '7z', 'zip', ... ]` list first ;)
Although I've just spotted `test/data/Patchwork-3.2.2-mac.dmg: bzip2 compressed data, block size = 100k` in your test results above, which I guess means that yeah it does need to check the file extension first, to prevent it mis-detecting a DMG as a BZIP2. Sorry.
But as I mentioned, do we really need to pull in all of `mime-types` just to determine whether a file ends with `.dmg` or not??